app/commands/v2/import.rb

```diff
@@ -0,0 +1,123 @@
+module Commands
+  module V2
+    class Import < BaseCommand
+      def call
+        unless UuidValidator.valid?(payload[:content_id])
+          raise CommandError.new(
+          code: 422,
+          error_details: {
+            error: {
+              code: 422,
+              message: "Content id not valid",
+              fields: "content_id",
+            }
+          })
+        end
+
+        delete_all(payload[:content_id])
+        all_content_items.map.with_index do |event, index|
+          create_content_item(event, index, payload[:content_id])
+        end
+
+        after_transaction_commit do
+          send_downstream(all_content_items.find { |e| e[:action] == 'Publish' })
+        end
+
+        Success.new(content_id: payload[:content_id])
+      end
+
+    private
+
+      def all_content_items
+        @all_content_items ||= [redirects.compact + payload[:content_items]].flatten
+      end
+
+      def send_downstream(content)
+        return unless content
+        DownstreamLiveWorker.perform_async_in_queue(
+          DownstreamLiveWorker::LOW_QUEUE,
+          content_id: content[:content_id],
+          locale: 'en',
+          message_queue_update_type: content[:payload][:update_type],
+          payload_version: event.id
```

Comment:
We should probably test for this if this is not covered, but I don't think this would be a problem. As the locking in the content store is done per base path, and providing that the Publishing API only sends one content item for each base path it wants to change (which I think is the behaviour), there should be no locking issues.